modules/custom/d_product/src/Plugin/views/argument/NodeVid.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Drupal\d_product\Plugin\views\argument;
+
+
+use Drupal\node\Plugin\views\argument\Vid;
+
+/**
+ * Argument handler to accept a node revision id.
+ *
+ * @ViewsArgument("node_vid")
+ */
+class NodeVid extends Vid {
+
+  /**
+   * Override the behavior of title(). Get the title of the revision.
+   */
+  public function titleQuery() {
+    $titles = [];
+
+    $results = $this->database->query('SELECT nr.vid, nr.nid, nfr.title 
+      FROM {node_revision} nr
+      INNER JOIN {node_field_revision} nfr ON nr.vid = nfr.vid
+      WHERE nr.vid IN ( :vids[] )', array(':vids[]' => $this->value))->fetchAllAssoc('vid', \PDO::FETCH_ASSOC);
+    $nids = [];
+    foreach ($results as $result) {
+      $nids[] = $result['nid'];
+    }
+
+    $nodes = $this->nodeStorage->loadMultiple(array_unique($nids));
+
+    foreach ($results as $result) {
+      $nodes[$result['nid']]->set('title', $result['title']);
+      $titles[] = $nodes[$result['nid']]->label();
+    }
+
+    return $titles;
```

Comment:
ok, nie do końca kumam, po co przypisujesz tu do `nodes[]->set('title'...` skoro potem znowu pobierasz ta labelkę i w sumie zwracasz z całej funkcji `$titles` ... Czy tu chodzi o jakieś "przetworzenie" tego tytułu przez wewnętrzne mechanizmy Drupala? wyjaśnij :pray: 

Jak nie ma wyraźnego powodu, to czy nie można by po prostu zwrócić wyniku zapytania na poczatku funkcji? 

```
$return $this->database->query('SELECT nr.vid, nr.nid, nfr.title 
      FROM {node_revision} nr
      INNER JOIN {node_field_revision} nfr ON nr.vid = nfr.vid
      WHERE nr.vid IN ( :vids[] )', array(':vids[]' => $this->value))->fetchAllKeyed(1, 2);
```
???

Reply:
Wziąłem po prostu rozwiązanie z patcha https://www.drupal.org/project/drupal/issues/2628130, skoro nie używamy patchy samych w sobie

Reply:
noo tak, ale kod wydaje się trochę dziwny.. możemy to poprawić skoro ostatecznie to nie patch, co sądzisz?

Reply:
Myślę że póki co można tak zrobić, jak się ujawnią jakieś problemy z viewsami korzystającymi z rewizji to można wtedy to bardziej analizować.